discovery-provider/src/tasks/index.py

```diff
@@ -354,6 +373,121 @@ def save_and_get_skip_tx_hash(session, redis):
     return None
 
 
+def group_transaction_by_type(tx_type_to_grouped_lists_map, tx, tx_receipt):
+    tx_target_contract_address = tx["to"]
+    for tx_type, tx_list in tx_type_to_grouped_lists_map.items():
+        tx_is_type = tx_target_contract_address == get_contract_addresses()[tx_type]
+        if tx_is_type:
+            contract_name = helpers.snake_to_pascal(tx_type)
+            logger.info(
+                f"index.py | {contract_name} contract addr: {tx_target_contract_address}"
+                f" tx from block - {tx}, receipt - {tx_receipt}"
+            )
+            tx_list.append(tx_receipt)
+            break
+
+
+def add_indexed_block_to_db(db_session, block):
+    web3 = update_task.web3
+    current_block_query = db_session.query(Block).filter_by(is_current=True)
+
+    # Without this check we may end up duplicating an insert operation
```

Comment:
I know you just copied this comment, but im not sure what this means in relation to this line, maybe 403 makes sense

Reply:
Yeah... I really don't know LOL; I'm going to look into this more. Thanks for flagging.